code/datums/extensions/state_machine.dm

```diff
@@ -0,0 +1,39 @@
+// This contains the current state of the FSM and should be held by whatever the FSM is controlling.
+// Unlike the individual states and their transitions, the state machine objects are not singletons, and hence aren't `/decl`s.
+/datum/extension/state_machine
+	expected_type = /datum
+	base_type = /datum/extension/state_machine
+	var/decl/state/current_state = null // Acts both as a ref to the current state and holds which state it will default to on init.
+
+/datum/extension/state_machine/New()
+	set_state(current_state)
+	return ..()
```

Comment:
New() shouldn't return a value I believe. Would also probably be better to call `..()` before any further logic since extensions have a bit of setup to do.

Reply:
I think I had `Initialize()` on the brain when I wrote that so I'll shove the `..()` above.